homeassistant/components/gios/air_quality.py

```diff
@@ -0,0 +1,166 @@
+"""Support for the GIOS service."""
+from homeassistant.components.air_quality import (
+    ATTR_CO,
+    ATTR_NO2,
+    ATTR_OZONE,
+    ATTR_PM_2_5,
+    ATTR_PM_10,
+    ATTR_SO2,
+    AirQualityEntity,
+)
+from homeassistant.const import CONF_NAME
+
+from .const import ATTR_STATION, DATA_CLIENT, DOMAIN, ICONS_MAP
+
+ATTRIBUTION = "Data provided by GIOŚ"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a GIOS entities from a config_entry."""
+    name = config_entry.data[CONF_NAME]
+
+    data = hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id]
+
+    async_add_entities([GiosAirQuality(data, name)], True)
+
+
+def round_state(func):
+    """Round state."""
+
+    def _decorator(self):
+        res = func(self)
+        if isinstance(res, float):
+            return round(res)
+        return res
+
+    return _decorator
+
+
+class GiosAirQuality(AirQualityEntity):
+    """Define an GIOS sensor."""
+
+    def __init__(self, gios, name):
+        """Initialize."""
+        self.gios = gios
+        self._name = name
+        self._aqi = None
+        self._co = None
+        self._no2 = None
+        self._o3 = None
+        self._pm_2_5 = None
+        self._pm_10 = None
+        self._so2 = None
+        self._icon = None
+        self._attrs = {}
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        if self._aqi in ICONS_MAP:
+            self._icon = ICONS_MAP[self._aqi]
+        else:
+            self._icon = "mdi:blur"
+        return self._icon
+
+    @property
+    def air_quality_index(self):
+        """Return the air quality index."""
+        return self._aqi
+
+    @property
+    @round_state
+    def particulate_matter_2_5(self):
+        """Return the particulate matter 2.5 level."""
+        return self._pm_2_5
+
+    @property
+    @round_state
+    def particulate_matter_10(self):
+        """Return the particulate matter 10 level."""
+        return self._pm_10
+
+    @property
+    @round_state
+    def ozone(self):
+        """Return the O3 (ozone) level."""
+        return self._o3
+
+    @property
+    @round_state
+    def carbon_monoxide(self):
+        """Return the CO (carbon monoxide) level."""
+        return self._co
+
+    @property
+    @round_state
+    def sulphur_dioxide(self):
+        """Return the SO2 (sulphur dioxide) level."""
+        return self._so2
+
+    @property
+    @round_state
+    def nitrogen_dioxide(self):
+        """Return the NO2 (nitrogen dioxide) level."""
+        return self._no2
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def state(self):
```

Comment:
Don't overwrite the `state` property. It's already implemented by the base air_quality class.